### PR TITLE
[GLUTEN-6612] Fix ParquetFileFormat issue caused by the setting of local property isNativeApplicable

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

* Fix typo: isNativeAppliable -> isNativeApplicable.
* Use == for null-safe comparing the value of local properties.
* Remove the check of this local property in `supportBatch` method. `supportBatch` is only called in read path, which shouldn't be impacted by that property used for native write.
* Reset local properties used for native write before next setting.

Fix https://github.com/apache/incubator-gluten/issues/6612, which shows inconsistent query behavior caused by historical property setting.

## How was this patch tested?

CI/local test.

### CODE REVIEW

File: shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala [hunk 1]
```diff
@@ -210,14 +210,10 @@ class ParquetFileFormat extends FileFormat with DataSourceRegister with Logging
 
   /** Returns whether the reader will return the rows as batch or not. */
   override def supportBatch(sparkSession: SparkSession, schema: StructType): Boolean = {
-    if ("true".equals(sparkSession.sparkContext.getLocalProperty("isNativeAppliable"))) {
```
Comment (PHILO-HE) [inline shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala:213]: @JkSelf, do you know why this check is added in the main code?

Comment (JkSelf) [reply shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala:213]: @PHILO-HE No need to add this check here. It is only used in parquet reader not writer.

File: shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala [hunk 2]
```diff
@@ -83,7 +83,7 @@ class OrcFileFormat extends FileFormat with DataSourceRegister with Serializable
       options: Map[String, String],
       files: Seq[FileStatus]): Option[StructType] = {
     // Why if (false)? Such code requires comments when being written.
-    if ("true".equals(sparkSession.sparkContext.getLocalProperty("isNativeAppliable")) && false) {
+    if ("true" == sparkSession.sparkContext.getLocalProperty("isNativeApplicable") && false) {
```
Comment (zhztheplayer) [inline shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala:86]: Would it be more straight forward to use `sparkSession.sparkContext.getLocalProperty("isNativeApplicable").toBoolean`?

**Review by zhztheplayer — APPROVED** [review]
I am still a little bit concerned about the use of these local properties. Perhaps we can manage to remove them later some time.

File: shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala (hunk 2 above)
Comment (PHILO-HE) [reply shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala:86]: @zhztheplayer, the property can be null and needs extra check if we use that. Maybe, just keep the current change.

> I am still a little bit concerned about the use of these local properties. Perhaps we can manage to remove them later some time.

Yes, it's not a good approach to use local property.

Comment (FelixYBW) [reply shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala:86]: Can we use true == xxx? why do we use string "true", is it from gluten config?

Comment (PHILO-HE) [reply shims/spark32/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala:86]: @FelixYBW, it's the raw value we obtained for this property.
The property value is kept by spark context and it is set by Gluten in `GlutenWriterColumnarRules`.

**Review by FelixYBW — APPROVED** [review]

**Comment by PHILO-HE** [comment]
The implementation for `inferSchema` has code like `if (xxx && false)`. I will confirm with this code's author to see whether we can remove it in another pr.

**Comment by FelixYBW** [comment]
> The implementation for `inferSchema` has code like `if (xxx && false)`. I will confirm with this code's author to see whether we can remove it in another pr.

Let's add comments in code once you get it. We should avoid such use.

**Pull request merged**